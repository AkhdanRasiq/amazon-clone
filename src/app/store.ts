import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import productReducer from '../features/productSlice'
import alertReducer from '../features/alertSlice'

export const store = configureStore({
  reducer: {
    products  : productReducer,
    alert     : alertReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
