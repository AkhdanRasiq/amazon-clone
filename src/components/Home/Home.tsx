import React from 'react'

import Product from '../Product/Product'

function Home() {
  return (
    <div className="home">
      <div className="homeContainer">
        <img className="homeBanner" src={require('../../assets/img/amazon_banner.jpg')} alt="amazon_banner" />

        <div className="homeRow">
          <Product
            a_strID="1"
            a_strTitle='The Product Title Props'
            a_iPrice={19.99}
            a_iRating={5}
            a_strImage="https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._AC_SY400_.jpg"
          />
          <Product
            a_strID="2"
            a_strTitle='Apple Pencil (2nd Generation)'
            a_iPrice={29.99}
            a_iRating={5}
            a_strImage="https://m.media-amazon.com/images/I/41S3MKU9TjL._AC_UL480_FMwebp_QL65_.jpg"
          />

        </div>

        <div className="homeRow">
          <Product
            a_strID="3"
            a_strTitle='SanDisk 128GB microSDXC Card, Licensed for Nintendo Switch - SDSQXAO-128G-GNCZN'
            a_iPrice={20.95}
            a_iRating={4}
            a_strImage="https://m.media-amazon.com/images/I/810o-i5jpiL._AC_UL480_FMwebp_QL65_.jpg"
          />
          <Product
            a_strID="4"
            a_strTitle='Dell USB 3.0 Ultra HD/4K Triple Display Docking Station (D3100), Black'
            a_iPrice={128.98}
            a_iRating={3}
            a_strImage="https://m.media-amazon.com/images/I/61qALmGqKGL._AC_UL480_FMwebp_QL65_.jpg"
          />
          <Product
            a_strID="5"
            a_strTitle='Logitech C922x Pro Stream Webcam â€“ Full 1080p HD Camera'
            a_iPrice={94.99}
            a_iRating={2}
            a_strImage="https://m.media-amazon.com/images/I/712xpaJPT6L._AC_UL480_FMwebp_QL65_.jpg"
          />

        </div>

        <div className="homeRow">
          <Product
            a_strID="6"
            a_strTitle='AMD Ryzen 5 5600X 6-core, 12-Thread Unlocked Desktop Processor with Wraith Stealth Cooler'
            a_iPrice={244.78}
            a_iRating={5}
            a_strImage="https://m.media-amazon.com/images/I/61vGQNUEsGL._AC_UL480_QL65_.jpg"
          />
        </div>
      </div>
    </div>
  )
}

export default Home
