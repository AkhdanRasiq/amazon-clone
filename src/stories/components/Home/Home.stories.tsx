import React from 'react'

import Home from '../../../components/Home/Home'
import '../../../assets/scss/styles.scss'

// import StoryRouter from 'storybook-react-router'
import { ComponentStory, ComponentMeta } from '@storybook/react'


export default {
  title: 'Components/Home',
  component: Home,
  // decorators: [StoryRouter()],
  argTypes: {
    submit: { action: 'submitted' }
  }
} as ComponentMeta<typeof Home>

const Template: ComponentStory<typeof Home> = (args: any) => <Home {...args} />


// const Template = function (args: any, { argTypes }: any) {
//   return (
//     <Home />
//   )
// }

export const Default: any = Template.bind({})
