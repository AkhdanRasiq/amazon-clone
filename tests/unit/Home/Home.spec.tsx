import React from 'react'
import { render } from "@testing-library/react"
import { store } from '../../../src/app/store';
import { Provider } from 'react-redux';

import Home from "../../../src/components/Home/Home"

describe("Home.tsx", () => {
  it("Render Properly", async () => {
    const wrapper = render(
      <Provider store={store}>
        <Home />
      </Provider>
    )
    await expect(wrapper).toMatchSnapshot()
  })
})
