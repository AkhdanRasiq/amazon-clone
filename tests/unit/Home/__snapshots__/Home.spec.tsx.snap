// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <div
          class="homeContainer"
        >
          <img
            alt="amazon_banner"
            class="homeBanner"
            src="[object Object]"
          />
          <div
            class="homeRow"
          >
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  The Product Title Props
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    19.99
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._AC_SY400_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  Apple Pencil (2nd Generation)
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    29.99
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://m.media-amazon.com/images/I/41S3MKU9TjL._AC_UL480_FMwebp_QL65_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
          </div>
          <div
            class="homeRow"
          >
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  SanDisk 128GB microSDXC Card, Licensed for Nintendo Switch - SDSQXAO-128G-GNCZN
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    20.95
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://m.media-amazon.com/images/I/810o-i5jpiL._AC_UL480_FMwebp_QL65_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  Dell USB 3.0 Ultra HD/4K Triple Display Docking Station (D3100), Black
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    128.98
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://m.media-amazon.com/images/I/61qALmGqKGL._AC_UL480_FMwebp_QL65_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  Logitech C922x Pro Stream Webcam Full 1080p HD Camera
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    94.99
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://m.media-amazon.com/images/I/712xpaJPT6L._AC_UL480_FMwebp_QL65_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
          </div>
          <div
            class="homeRow"
          >
            <div
              class="product"
            >
              <div
                class="productInfo"
              >
                <p>
                  AMD Ryzen 5 5600X 6-core, 12-Thread Unlocked Desktop Processor with Wraith Stealth Cooler
                </p>
                <p
                  class="productPrice"
                >
                  <small>
                    $
                  </small>
                  <strong>
                    244.78
                  </strong>
                </p>
                <div
                  class="productRating"
                >
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                  <p>
                    ⭐
                  </p>
                </div>
              </div>
              <img
                alt="product_img"
                src="https://m.media-amazon.com/images/I/61vGQNUEsGL._AC_UL480_QL65_.jpg"
              />
              <button>
                Add to Basket
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <div
        class="homeContainer"
      >
        <img
          alt="amazon_banner"
          class="homeBanner"
          src="[object Object]"
        />
        <div
          class="homeRow"
        >
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                The Product Title Props
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  19.99
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://images-na.ssl-images-amazon.com/images/I/51Zymoq7UnL._AC_SY400_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                Apple Pencil (2nd Generation)
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  29.99
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://m.media-amazon.com/images/I/41S3MKU9TjL._AC_UL480_FMwebp_QL65_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
        </div>
        <div
          class="homeRow"
        >
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                SanDisk 128GB microSDXC Card, Licensed for Nintendo Switch - SDSQXAO-128G-GNCZN
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  20.95
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://m.media-amazon.com/images/I/810o-i5jpiL._AC_UL480_FMwebp_QL65_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                Dell USB 3.0 Ultra HD/4K Triple Display Docking Station (D3100), Black
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  128.98
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://m.media-amazon.com/images/I/61qALmGqKGL._AC_UL480_FMwebp_QL65_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                Logitech C922x Pro Stream Webcam Full 1080p HD Camera
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  94.99
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://m.media-amazon.com/images/I/712xpaJPT6L._AC_UL480_FMwebp_QL65_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
        </div>
        <div
          class="homeRow"
        >
          <div
            class="product"
          >
            <div
              class="productInfo"
            >
              <p>
                AMD Ryzen 5 5600X 6-core, 12-Thread Unlocked Desktop Processor with Wraith Stealth Cooler
              </p>
              <p
                class="productPrice"
              >
                <small>
                  $
                </small>
                <strong>
                  244.78
                </strong>
              </p>
              <div
                class="productRating"
              >
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
                <p>
                  ⭐
                </p>
              </div>
            </div>
            <img
              alt="product_img"
              src="https://m.media-amazon.com/images/I/61vGQNUEsGL._AC_UL480_QL65_.jpg"
            />
            <button>
              Add to Basket
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
